Setup: Feature/UI-Design Branch for Vanilla To-Do App with Server

codespirityash opened the pull request:
### Summary  
This pull request sets up the `feature/ui-design` branch for the Vanilla To-Do App with server functionality. The UI will be designed using HTML and SCSS to ensure a clean and maintainable structure.  

### Next Steps  
The branch will be used to develop and refine the user interface, including task input, display, and interaction components.

codespirityash committed:
setting up css

yashweblife commented on client/src/styles/index.scss:
```diff
@@ -0,0 +1,6 @@
+do 
```
remove do, its not valid css

codespirityash committed:
remove invalid css

codespirityash reviewed (commented):
Removed do ,as required in the review of previous commit

codespirityash committed:
Merge branch 'main' into feature/ui-design

codespirityash committed:
create html structure for todo

codespirityash reviewed (commented):
This is a html structure for the todo list app's UI Design

yashweblife commented on client/index.html:
```diff
@@ -7,7 +7,35 @@
+          <div class="header"><h4>Task Title 1</h4></div>
+          <div class="content">Task description for Task 1</div>
+          <div class="footer">
+            <button id="edit">Edit</button>
```
instead if edit, add a `complete` button

yashweblife commented on client/index.html:
```diff
@@ -7,7 +7,35 @@
+        </section>
+      </section>
+      <footer>
+        <input type="text" id="input" placeholder="What do you have in your mind?">
```
The placeholder seems too big, "What's on your mind" is short and sweet

yashweblife commented on client/index.html:
```diff
@@ -7,7 +7,35 @@
+        <input type="text" id="input" placeholder="What do you have in your mind?">
+        <button id="button">Add Task</button>
+      </footer>
+      </footer>
```
There are 2 footer closing tags for some reason

yashweblife committed:
add more ignores

yashweblife commented on client/index.html:
```diff
@@ -7,7 +7,35 @@
+      <header>
+        <h2>TO-DO APP WITH SERVER</h2>
+      </header>
+      <section id="Task-list">
```
`Task-list` is not a good way to name
Use `task-list` or `tasklist` or `Task-List`
Always be consistent with your naming conventions
For JS: use `variableName` or `VariableName` or `variable_name`
For HTML: best to use `tag-name` or `tagname`

codespirityash committed:
Resolve HTML issues

codespirityash committed:
Add Css Styling to HTML Structure

codespirityash committed:
Add Functioality with DOM in js

codespirityash commented:
Added Functionality using Javascript and DOM

yashweblife commented on client/index.html:
```diff
@@ -4,10 +4,20 @@
+        <button id="button">Add Task</button>
+      </footer>
+    </div>
+    <script type="module" src="./src/lib/index.js"></script>
```
This should point to `main.js`, not `./src/lib/index.js`

yashweblife commented on client/index.html:
```diff
@@ -4,10 +4,20 @@
     <meta charset="UTF-8" />
     <link rel="icon" type="image/svg+xml" href="/vite.svg" />
     <meta name="viewport" content="width=device-width, initial-scale=1.0" />
+    <link rel="stylesheet" href="\src\styles\index.scss" />
```
This wont compile during the build process. Add an import statement to `main.js`:`import "./styles/index.scss";`.
That tells the vite bundler to compile scss into css for us. Scss is a css preprocessor, not a replacement for css so we need to convert scss to css using a preprocessor/bundler.

yashweblife commented on client/index.html:
```diff
@@ -4,10 +4,20 @@
     <link rel="icon" type="image/svg+xml" href="/vite.svg" />
     <meta name="viewport" content="width=device-width, initial-scale=1.0" />
+    <link rel="stylesheet" href="\src\styles\index.scss" />
     <title>Vite App</title>
```
Change the name in title, this tag represents the name you see on the browser tab(top bar)

yashweblife commented on client/src/lib/index.js:
```diff
@@ -0,0 +1,61 @@
+document.addEventListener("DOMContentLoaded", function () {
```
put this code into main.js
lib folder is reserver for libraries only. they shouldnt directly interact with pages. Use the import export syntax.
If you made a special reuseable(can be used in many places) function, put it in lib.
Also you dont need this event listener, it will work if its in the main.js file

yashweblife commented on client/src/styles/index.scss:
```diff
@@ -0,0 +1,109 @@
+@import "./resets";
```
import statements are depricated in scss. dont use them.
Instead try using [this](https://sass-lang.com/guide/#modules)

yashweblife commented:
All said, this is looking pretty good, we will merge this into main later, but we need to make it modern looking and use dark theme by default. We also want to make the code more robust so that it can tolerate having a server connected to it. I would suggest making a class to create and manage DOM stuff. I would also suggest making smaller self contained functions instead of large monolithic functions. That way we will be able to test our code too later.

yashweblife commented:
Maybe use this color palette:
[Link](https://coolors.co/12100e-30321c-4a4b2f-6b654b-d4df9e)
Or this palette:
```scss
/** Dark theme primary colors */
$color-primary-a0: #38d6c4;
$color-primary-a10: #5cdbca;
$color-primary-a20: #76e0d1;
$color-primary-a30: #8de5d7;
$color-primary-a40: #a2e9de;
$color-primary-a50: #b6eee4;

/** Dark theme surface colors */
$color-surface-a0: #241e22;
$color-surface-a10: #393337;
$color-surface-a20: #4e494c;
$color-surface-a30: #656063;
$color-surface-a40: #7d797b;
$color-surface-a50: #969294;

/** Dark theme mixed surface colors */
$color-surface-mixed-a0: #3c746d;
$color-surface-mixed-a10: #52837c;
$color-surface-mixed-a20: #67918b;
$color-surface-mixed-a30: #7ca19b;
$color-surface-mixed-a40: #91b0ab;
$color-surface-mixed-a50: #a7bfbb;

 /** Examples */ 
.bg-primary {
 color: $color-primary-a50; 
 background-color: $color-surface-a0; 
}
```

codespirityash committed:
Organize Code

codespirityash commented:
The code has been organized.

-Replaced import statements with @use in main.scss
-Added a reusable function in lib
-Moved the code that manipulates Dom to main.js

Points yet to work on:-

1)Building a class to manage DOM
2)Optimize code by adding smaller functions
3)Change bg-color using a decent color pallete
